src/configuration.ts

```diff
@@ -208,46 +210,36 @@ export class ConnectionConfiguration implements IConnectionConfiguration {
   }
 
   get pdkRubyVerDir(): string {
-    var rootDir = path.join(this.puppetBaseDir, 'private', 'puppet', 'ruby');
-
-    return this.findFirstDirectory(rootDir);
+    this.getPDKInstance();
```

Comment:
Instead of having to call `this.getPDKInstance()` each time for each property, can the contents of `getPDKInstance` be put inside the constructor?

Reply:
IMO, I try to keep as much "magic" as possible out of object constructors.  Makes it easier to overload methods or change things.  Also makes testing easier.  I class detecting PDK instances as "magical"

I kept the original methods e.g. `pdkRubyVerDir` etc. to keep the amount of change to a minimum, but  we could solve this in one of two ways.

1. Remove these "legacy" methods and create a new method called `pdkInstance` that returns an IPDKInstance object. Basically handing off the repsonsibility to the object consumers

2. I could use a oneliner instead for all of the "legacy" methods.
`return this.getPDKInstance().rubyVerDir;`

Reply:
After discussion, I think this is fine as is now. If it should change, the next tickets to use this to allow user to change version should refine it.